x-pack/plugins/encrypted_saved_objects/server/routes/key_rotation.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { RouteDefinitionParams } from '.';
+
+/**
+ * The default maximum value of from + size for searches to .kibana index. Since we cannot use scroll
+ * or search_after functionality with the .kibana index we limit maximum batch size with this value.
+ */
+const DEFAULT_MAX_RESULT_WINDOW = 10000;
```

Comment:
__note:__ we could also fetch this index setting here and know the exact number, but I guess we don't want users to change index settings for our own internal indices so it should be relative safe to assume defaults here.

Reply:
Thanks for the details, do you think we shouldn't restrict `batchSize` parameter with a max 10000, and only use this value as a default?

Reply:
> Thanks for the details, do you think we shouldn't restrict batchSize parameter with a max 10000, and only use this value as a default?

No, I think what you have is good. We can always revisit this, but maxing at 10000 makes sense to me. We don't want to overload Kibana